samples/src/test/java/org/autorefactor/refactoring/rules/samples_in/BooleanSample.java

```diff
@@ -147,24 +147,101 @@ public boolean returnIfConditionBooleanObject4(boolean b) {
         return Boolean.TRUE;
     }
 
-    public void removeUselessUseOfTernaryOperatorWithBooleanPrimitive1(
-            boolean bo) {
+    public void removeUselessTernaryOperatorWithBooleanPrimitive1(boolean bo) {
         boolean b = bo ? true : false;
     }
 
-    public void removeUselessUseOfTernaryOperatorWithBooleanPrimitive2(
-            boolean bo) {
+    public void removeUselessTernaryOperatorWithBooleanPrimitive2(boolean bo) {
         boolean b = bo ? false : true;
     }
 
-    public void removeUselessUseOfTernaryOperatorWithBooleanObject1(boolean bo) {
+    public void removeUselessTernaryOperatorWithBooleanObject1(boolean bo) {
         Boolean b = bo ? Boolean.TRUE : Boolean.FALSE;
     }
 
-    public void removeUselessUseOfTernaryOperatorWithBooleanObject2(boolean bo) {
+    public void removeUselessTernaryOperatorWithBooleanObject2(boolean bo) {
         Boolean b = bo ? Boolean.FALSE : Boolean.TRUE;
     }
 
+    public void replaceTernaryOperatorByAndOperator(boolean bo1, boolean bo2) {
+        boolean b = bo1 ? bo2 : false;
+    }
+
+    public void replaceTernaryOperatorByAndOperator2(boolean bo1, boolean bo2) {
+        boolean b = bo1 ? false : bo2;
+    }
+
+    public void replaceTernaryOperatorByOrOperator(boolean bo1, boolean bo2) {
+        boolean b = bo1 ? true : bo2;
+    }
+
+    public void replaceTernaryOperatorByOrOperator2(boolean bo1, boolean bo2) {
+        boolean b = bo1 ? bo2 : true;
+    }
+
+    public void replaceTernaryOperatorByAndOperatorWithObjectConstant(
+            boolean bo1, boolean bo2) {
+        Boolean b = bo1 ? bo2 : Boolean.FALSE;
+    }
+
+    public void replaceTernaryOperatorByAndOperatorWithObjectConstant2(
+            boolean bo1, boolean bo2) {
+        Boolean b = bo1 ? Boolean.FALSE : bo2;
+    }
+
+    public void replaceTernaryOperatorByOrOperatorWithObjectConstant(
+            boolean bo1, boolean bo2) {
+        Boolean b = bo1 ? Boolean.TRUE : bo2;
+    }
+
+    public void replaceTernaryOperatorByOrOperatorWithObjectConstant2(
+            boolean bo1, boolean bo2) {
+        Boolean b = bo1 ? bo2 : Boolean.TRUE;
+    }
+
+    public void replaceTernaryOperatorByAndOperatorWithObject(
+            Boolean bo1, Boolean bo2) {
+        Boolean b = bo1 ? bo2 : false;

```

Comment:
We must avoid the case where both operands are objects:
- 2 primitives -> no null pointer
- 1 primitive + 1 object -> null pointer with and without refactoring (no regression)
- 2 objects -> no null pointer without refactoring / null pointer with refactoring (regression)

Let the following code be:
```Java
Boolean myBoolean = null;
Boolean myNewBoolean = (1 == 1) ? myBoolean : false;
```

You get a null pointer because you have one operand that is a primitive so the other operand is first casted as a primitive but it is null so you have a null pointer.